includes/compat/class-wc-stripe-subs-compat.php

```diff
@@ -274,12 +283,12 @@ public function process_subscription_payment( $amount = 0.0, $renewal_order, $re
 				$order_id = WC_Stripe_Helper::is_wc_lt( '3.0' ) ? $renewal_order->id : $renewal_order->get_id();
 
 				WC_Stripe_Helper::is_wc_lt( '3.0' ) ? update_post_meta( $order_id, '_transaction_id', $id ) : $renewal_order->set_transaction_id( $id );
-				$renewal_order->update_status( 'on-hold', sprintf( __( 'Stripe charge awaiting authentication by user: %s.', 'woocommerce-gateway-stripe' ), $id ) );
+				$renewal_order->update_status( 'failed', sprintf( __( 'Stripe charge awaiting authentication by user: %s.', 'woocommerce-gateway-stripe' ), $id ) );
```

Comment:
@v18 I changed the order status to `failed` instead of `on-hold`. The main reason is that, when SCA fails, a webhook of type `charge.failed` is emitted. That webhook simply does nothing if the order status is already `failed`. If it's not `failed`, the webhook sets it to `failed` (which creates an order note, and another email sent to the customer). We could guard against this specific scenario in the webhook handler, but I think marking the order here as `failed` is the simplest solution. Since "failed" is not an irreversible status, it kind-of fits semantically: The payment has failed until the customer performs an action to fix it. In that way, it's not different than if the credit card has no funds temporarily, or it has expired.

I've confirmed that the subscription is set to `on-hold` when the renewal order is set to `failed`. This change will also make @dwainm's integration on wc.com a bit easier, because they don't use the `on-hold` status anywhere, but they do use `failed`.

Reply:
Thank you @DanReyLop